### PR TITLE
ci: fix circleci workflow

### PR DESCRIPTION
**Pull request by superus8r**
Relates OR Closes #0000

## Description
CircleCI seems to have fixed an account issue which did not allow starting jobs with macOS executor
- update the workflow to match circleCI's new update regarding macOS executors
- fix 0% sonar coverage useing generic xml format for uploading coverage files to sonar
- fix an issue with Codecov showing coverage on wrong files
- fix an issue with coverage files generated for wrong scheme
- separate codecov upload job from the build-and-test-job for simplicity
- add CircleCI build badge to README

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -20,7 +20,7 @@ platform :ios do
   lane :unit_tests do
     # add actions here: https://docs.fastlane.tools/actions
     test_args = {
-        scheme: "TrustylibUnitTests",
+        scheme: "Trustylib",
         device: "iPhone 11",
```
Comment (prempratapsingh) [inline fastlane/Fastfile:24]: Unit tests are linked to the `TrustylibUnitTests` xcode schme, setting the scheme to `Trustylib` may not execute  the unit tests.

Can you check please if the unit tests are running with `bundle exec fastlane unit_tests` command?

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,75 +3,75 @@ orbs:
   codecov: codecov/codecov@3.2.4
   sonarcloud: sonarsource/sonarcloud@1.1.1
 
+executors:
+  linux-docker:
+    docker:
+      - image: cimg/base:2023.03
+
 jobs:
```
Comment (prempratapsingh) [inline .circleci/config.yml:11]: Its a great news that the CircleCI build issue is finally fixed 👍
Does adding `linux-docker` as `executors` fix the problem?

**Review by prempratapsingh — COMMENTED** [review]
It sounds great that the CircleCI build issue can finally be fixed 🚀

File: fastlane/Fastfile (hunk 1 above)
Comment (superus8r) [reply fastlane/Fastfile:24]: That is correct 🙂  the unit tests still remain linked to `TrustylibUnitTests`.
The change of scheme here fixes an issue with 0 percent coverage on Sonar and coverage on test files on codecov.
Now Sonar will show the correct coverage percent and codecov shows the generated coverage by tests for `Trustylib` scheme.

> Can you check please if the unit tests are running with bundle exec fastlane unit_tests command?

Yes 🙂 The unit tests are running with `bundle exec fastlane unit_tests` command:
https://app.circleci.com/pipelines/github/superus8r/etrusted-ios-trustbadge-library/25/workflows/61191f03-9aa4-49ec-9bc7-3275eb2147c2/jobs/43

File: .circleci/config.yml (hunk 2 above)
Comment (superus8r) [reply .circleci/config.yml:11]: Thank you, it is indeed great news that CircleCI finally fixed macOS executors with our accounts.
> Does adding linux-docker as executors fix the problem?

Adding `linux-docker` as `executors` helps to reuse the executor for `sonar_analysis` and `codecov_upload` jobs (so the short answer would be no).

The executor for `build-and-test` which requires macOS remains `macos.x86.medium.gen2`, but the problem with running macOS executors is fixed externally by CircleCI. 🙂

File: fastlane/Fastfile (hunk 1 above)
Comment (prempratapsingh) [reply fastlane/Fastfile:24]: Awesome! thanks for confirming

**Review by prempratapsingh — APPROVED** [review]
Looks great 🚀

**Pull request merged**